### ISSUE TITLE
Different behaviors between Tendermint 0.14.0 and Tendermint 0.15.0 with abci-cli and Ethermint.

### ISSUE THREAD

**Issue by MarcoFavorito** [tendermint/tendermint#1172]
**BUG REPORT**

**Tendermint versions**
1) `0.15.0-381fe19` (installed with APT)
2) `0.14.0-88f5f21` (installed from source)

**ABCI apps**:
1) `abci-cli dummy`, `abci-cli` version 0.9.0
2) `ethermint`:
	
		$GOPATH/bin/ethermint version
		ethermint:  0.5.4-e7c9e10
		go-ethereum:  1.6.7-stable

**Environment**:
- **OS** (e.g. from /etc/os-release):

		NAME="Ubuntu"
		VERSION="16.04.3 LTS (Xenial Xerus)"
		ID=ubuntu
		ID_LIKE=debian
		PRETTY_NAME="Ubuntu 16.04.3 LTS"
		VERSION_ID="16.04"
		HOME_URL="http://www.ubuntu.com/"
		SUPPORT_URL="http://help.ubuntu.com/"
		BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
		VERSION_CODENAME=xenial
		UBUNTU_CODENAME=xenial

**What happened**:
In short:
1. with `abci-cli dummy`:
	1. `tendermint 0.15.0-381fe19` works
	2. `tendermint 0.14.0-88f5f21` does not work

2. with `ethermint` __the opposite__:
	1. `tendermint 0.15.0-381fe19` does not works
	2. `tendermint 0.14.0-88f5f21` works

**TL;DR**:

- This sreenshot is for the **case 1** (i.e. with `abci-cli dummy`)
![image](https://user-images.githubusercontent.com/9467617/35518455-ee33f9b6-0511-11e8-8382-d42cfc279863.png)

**Notice** that in the case 1.2 stuff works if I run `tendermint node --proxy_app dummy`

- This one is for the **case 2.1** (i.e. with `ethermint` and `tendermint 0.15.0-381fe19`)
![image](https://user-images.githubusercontent.com/9467617/35518743-adec8af2-0512-11e8-9990-2ca42d76c238.png)

- This one is for the **case 2.2** (i.e. with `ethermint` and `tendermint 0.14.0-88f5f21`)
![image](https://user-images.githubusercontent.com/9467617/35518886-1488698e-0513-11e8-9f36-8cd4ba7bea01.png)

**What you expected to happen**:
That works in all cases, or at least that with the last version of `tendermint` it works with both `abci-cli` and `ethermint`.

**How to reproduce it** (as minimally and precisely as possible):
Run the commands on the above screenshots. I always run first the app, but it should be irrelevant. 
If you are in doubt, please look at the times in the outputs. 
If they are not clear, please tell me. Thank you.

**Comment by zramsay** [comment]
we're working on documenting version (in)compatibilities, see https://github.com/tendermint/aib-data/issues/31

since you can get it working and the incompatibilities are from breaking changes that haven't percolated through the stack, rather than a bug, I'll close this issue; but it will be good information when working on the above issue
